### PR TITLE
Monday's homework from LG

### PR DESCRIPTION
**Pull request by MojoBoot**
Well, I understand the json stuff a lot better that I did, but still a ways to go...it's powerful and I can't actually believe it can do so much!

happy monday

### CODE REVIEW

**Comment by tgaff** [comment]
hi @MojoBoot would you mind moving your files into directories for each homework?  eg. 'hw_4', 'hw_5'

**Comment by MojoBoot** [comment]
👍

**Comment by MojoBoot** [comment]
Files have been moved and robot class work added. Should I make another pull request?

File: LeslieGomez-MojoBoot/HW- 6/rise_of_robot.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+####### alternate version from Kisha ####################
+# def self.make_robots(number)                          #
+#   robots = number.to_i                                #
+#   robots.times do                                     #
+#     robot= Robot.new("Max", "X500", "San Francisco")  #
+#     puts "#{robot.name} has been created"             #
+#   end                                                 #
+# end                                                   #
+#########################################################
```
Comment (Borgaard) [inline LeslieGomez-MojoBoot/HW- 6/rise_of_robot.rb]: :+1:

File: LeslieGomez-MojoBoot/HW- 6/rise_of_robot.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+def self.random_robot_maker
+names = ["Celeste", "Max", "Ruby", "Ella", "Prince"]
+type = ["Android", "Super-Android"]
+origin = ["Honolulu", "London", "San Francisco", "Dubai", "Barcelona"]
+
+n = names.sample
+t = type.sample
+o = origin.sample
+
+robot = Robot.new(n,t,o)
```
Comment (Borgaard) [inline LeslieGomez-MojoBoot/HW- 6/rise_of_robot.rb]: It seems you're grasping Classes really well. I do want to point out a best practice of Ruby, which is that it should read like English. A variable should be a concise description of what it holds. `n` holds samples of names, yes, but I'd have to poke through your code to figure that out.

File: LeslieGomez-MojoBoot/HW- 6/rise_of_robot.rb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+#constructor -this initialize method creates new instances of the Robot class
+#includes the instance variables required and initializes them
+def initialize(name, model, origin)
+  @name = name
+  @model = model
+  @origin = origin
+end
+#@instance-variable are available throughout the class
+#@name is an an instance variable
+
```
Comment (Borgaard) [inline LeslieGomez-MojoBoot/HW- 6/rise_of_robot.rb]: Clear commenting is a great way to learn in the moment, and remind future Leslie what this code is doing.

File: LeslieGomez-MojoBoot/HW- 5/leslie_mashable_get_it.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  front_page(new_story)
+end
+
+def front_page(new_story)
+  frontpage = []
+  frontpage << new_story
+  puts "Headlines from #{new_story[:category]}:"
+  puts "#{new_story[:title]}...This story has been shared #{new_story[:shares]} times. For more information go to #{new_story[:link]}}"
+  puts
+end
```
Comment (Borgaard) [inline LeslieGomez-MojoBoot/HW- 5/leslie_mashable_get_it.rb]: Oh nice handling of popular stories!

**Comment by MojoBoot** [comment]
Thank you for the feedback

**Pull request merged**